adapters/ddf/src/main/java/com/connexta/replication/adapters/ddf/DdfNodeAdapter.java

```diff
@@ -0,0 +1,569 @@
+              ecqlFilter,
+              filterBuilder.anyOf(
+                  filterBuilder.allOf(filters), filterBuilder.allOf(deletedFilters)));
+    } else {
+      filters.add(ecqlFilter);
+      finalFilter = filterBuilder.allOf(filters);
+    }
+
+    if (!failedItemFilters.isEmpty()) {
+      finalFilter = filterBuilder.anyOf(finalFilter, filterBuilder.anyOf(failedItemFilters));
+    }
+
+    Query query =
+        new QueryImpl(
+            finalFilter,
+            1,
+            100,
+            new SortByImpl(Core.METACARD_MODIFIED, SortOrder.ASCENDING),
+            false,
+            0L);
+    ddf.catalog.operation.QueryRequest request = new QueryRequestImpl(query);
+
+    ResultIterable results = ResultIterable.resultIterable(super::query, request);
+    return new QueryResponseImpl(new ResultIterableWrapper(results));
+  }
+
+  @Override
+  public boolean exists(Metadata metadata) {
+    final String metacardId = metadata.getId();
+    try {
+      return this.query(
+                  new QueryRequestImpl(
+                      new QueryImpl(
+                          filterBuilder.attribute(Core.ID).is().equalTo().text(metacardId))))
+              .getHits()
+          > 0;
+    } catch (UnsupportedOperationException | UnsupportedQueryException e) {
+      throw new AdapterException(
+          String.format(
+              "Error checking for the existence of metacard %s on %s",
+              metacardId, getSystemName()));
+    }
+  }
+
+  @Override
+  public boolean createRequest(CreateRequest createRequest) {
+    List<Metacard> metacards = getMetacards(createRequest.getMetadata());
+
+    try {
+      CreateResponse createResponse = super.create(new CreateRequestImpl(metacards));
+      checkForProcessingErrors(createResponse, "CreateResponse");
+    } catch (IngestException e) {
+      LOGGER.debug("Failed to create metacards to remote store", e);
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean updateRequest(UpdateRequest updateRequest) {
+    List<Metacard> metacards = getMetacards(updateRequest.getUpdatedMetadata());
+    String[] ids = new String[metacards.size()];
+
+    for (int i = 0; i < metacards.size(); i++) {
+      ids[i] = metacards.get(i).getId();
+    }
+
+    // assume content uri scheme since we cannot fetch the original resource uri over csw since it
+    // gets rewritten. This gets around the update getting denied by the framework thinking we are
+    // trying to change the resource uri, which is not allowed.
+    metacards.forEach(
+        metacard ->
+            metacard.setAttribute(
+                new AttributeImpl(
+                    Core.RESOURCE_URI, ContentItem.CONTENT_SCHEME + ":" + metacard.getId())));
+
+    ddf.catalog.operation.UpdateRequest ddfUpdate = new UpdateRequestImpl(ids, metacards);
+
+    // todo: This previous state metacards is not correct, but it was this way before. Does this
+    // have any effect?
```

Comment:
The CswStore doesn't do anything with these metacards it just hands them back to us in the response but the request will fail if you don't include them. 🙁 So we add this property just to keep the store happy.